lib/pure/uri.nim

```diff
@@ -165,18 +165,21 @@ iterator decodeQuery*(data: string): tuple[key, value: TaintedString] =
   ## data consists of.
   runnableExamples:
     import std/sugar
-    let s = collect(newSeq):
-      for k, v in decodeQuery("foo=1&bar=2"): (k, v)
-    doAssert s == @[("foo", "1"), ("bar", "2")]
+    let s = collect:
+      for k, v in decodeQuery("foo=1&bar=2=3"): (k, v)
+    doAssert s == @[("foo", "1"), ("bar", "2=3")]
```

Comment:
also add this:
```nim
doAssert toSeq(decodeQuery("&a&=b&=&&")) == @[("", ""), ("a", ""), ("", "b"), ("", ""), ("", "")]
```

Reply:
Please don't fill the runnable examples with dozens of corner cases. They're meant to show how to use the procedure, not every single result that is possible.

Put these into a `when isMainModule` or another test file.

Reply:
We actually have a test file for this already: https://github.com/nim-lang/Nim/blob/devel/tests/stdlib/turi.nim. Please add these tests in there.

Reply:
it's not dozens of corner cases, it's literally 2 examples. And yes, when I read docs, I do care about how are corner cases handled, otherwise you end up doing detective by reading the implementation to figure out the behavior, which defeats purpose of having runnableExamples in the 1st place.